server/xpub-client/pubsub.test.js

```diff
@@ -0,0 +1,118 @@
+jest.useFakeTimers()
+const PubsubClient = require('./pubsub.js')
+
+const createPubsubManager = mockGet => ({ getPubsub: mockGet })
+const createPubsub = (mockGetPubsub, mockPubsub = {}) => {
+  const mockPubsubManager = createPubsubManager(
+    mockGetPubsub || (() => mockPubsub),
+  )
+  return new PubsubClient(mockPubsubManager)
+}
+
+describe('PubsubClient', () => {
+  describe('initialize', () => {
+    it('calls getPubsub on the pubsub manager and stores it as an internal value', async () => {
+      const mockPubsub = {}
+      const mockGetPubsub = jest.fn(() => mockPubsub)
+      const pubsub = createPubsub(mockGetPubsub)
+      await pubsub.initialize()
+
+      expect(mockGetPubsub).toBeCalled()
+      expect(pubsub.pubsub).toBe(mockPubsub)
+    })
+  })
+
+  describe('publish', () => {
+    it('calls pubsub.publish and uses the callback to get the message value', async () => {
```

Comment:
These tests relate to the start and stop interval. maybe a new test for publish would simply check it calls `publish` on its `pubsub` property and correctly forwards the params on?